### ISSUE TITLE
precompiled gem & trunk

### ISSUE THREAD

**Issue by MSP-Greg** [sparklemotion/nokogiri#1839]
RubyGems has issues with ruby version constraints and proper gem selection when pre-compiled (platform specific) gems are available.

Ignoring that, the upper ruby version constraint used for `nokogiri-1.8.5-x64-mingw32.gem` is "< 2.6", which I believe will allow 'dev' and 'pre' builds to 'pass', and hence, the gem is installed.  Maybe use a constraint of "<= 2.5.99" (or "<= 2.6.99" in a few weeks)?

The following occurs with a trunk install:
```
PS C:\> ruby -v
ruby 2.6.0dev (2018-12-13 trunk 66384) [x64-mingw32]

PS C:\> gem install nokogiri
Fetching nokogiri-1.8.5-x64-mingw32.gem
Nokogiri is built with the packaged libraries: libxml2-2.9.8, libxslt-1.1.32, zlib-1.2.11, libiconv-1.15.
Successfully installed nokogiri-1.8.5-x64-mingw32
1 gem installed

PS C:\> nokogiri -v
Traceback (most recent call last):
        7: from C:/Greg/Ruby99-x64/bin/nokogiri:23:in `<main>'
        6: from C:/Greg/Ruby99-x64/bin/nokogiri:23:in `load'
        5: from C:/Greg/Ruby99-x64/lib/ruby/gems/2.6.0/gems/nokogiri-1.8.5-x64-mingw32/bin/nokogiri:6:in `<top (required)>'
        4: from C:/Greg/Ruby99-x64/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        3: from C:/Greg/Ruby99-x64/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        2: from C:/Greg/Ruby99-x64/lib/ruby/gems/2.6.0/gems/nokogiri-1.8.5-x64-mingw32/lib/nokogiri.rb:30:in `<top (required)>'
        1: from C:/Greg/Ruby99-x64/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
C:/Greg/Ruby99-x64/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- nokogiri/2.6/nokogiri (LoadError)
```

**Comment by larskanis** [comment]
Good point @MSP-Greg . The ruby version restriction is actually set in  rake-compiler here: https://github.com/rake-compiler/rake-compiler/blob/master/lib/rake/extensiontask.rb#L263

Are you sure the ruby patch version will not exceed 99?

**Comment by MSP-Greg** [comment]
@larskanis 

> Are you sure the ruby patch version will not exceed 99?

Oops.  Came across the issue while frustrated with something else.  I suppose both of us could win the lotto's, but unlikely...

I think the better constraint would be `"< 2.6.0.a"`.

I've never built any of the Ruby 'pre-releases', so I can't test with them, but I think the 'a' should filter correctly.

> set in rake-compiler here

Thanks.  Never looked at the testing there, so...

Happy Holidays,  Greg

**Comment by MSP-Greg** [comment]
Moved to [rake-compiler issue 150](https://github.com/rake-compiler/rake-compiler/issues/150)

Ok to close...

**Comment by MSP-Greg** [comment]
See [rake-compiler/rake-compiler 6292719](https://github.com/rake-compiler/rake-compiler/commit/6292719d9a0eed8d3708d929060c41b103b51942), included in 1.0.6.
